modules/externalipam/phpipam/phpipam_client.rb

```diff
@@ -0,0 +1,188 @@
+require 'externalipam/ipam_validator'
+require 'externalipam/api_resource'
+require 'externalipam/ip_cache'
+
+module Proxy::Phpipam
+  # Implementation class for External IPAM provider phpIPAM
+  class PhpipamClient
+    include Proxy::Log
+    include Proxy::Ipam::IpamHelper
+    include Proxy::Ipam::IpamValidator
+
+    def initialize(conf)
+      @conf = conf
+      @api_base = "#{@conf[:url]}/api/#{@conf[:user]}/"
+      @token = authenticate
+      @api_resource = Proxy::Ipam::ApiResource.new(api_base: @api_base, token: @token, auth_header: 'Token')
+      @ip_cache = Proxy::Ipam::IpCache.instance
+      @ip_cache.provider_name('phpipam')
+    end
+
+    def get_ipam_subnet(cidr, group_name = nil)
+      if group_name.nil? || group_name.empty?
+        get_ipam_subnet_by_cidr(cidr)
+      else
+        group = get_ipam_group(group_name)
+        get_ipam_subnet_by_group(cidr, group[:id])
+      end
+    end
+
+    def get_ipam_subnet_by_group(cidr, group_id)
+      subnets = get_ipam_subnets(group_id)
+      return nil if subnets.nil?
+      subnet_id = nil
+
+      subnets.each do |subnet|
+        subnet_cidr = "#{subnet[:subnet]}/#{subnet[:mask]}"
+        subnet_id = subnet[:id] if subnet_cidr == cidr
+      end
+
+      return nil if subnet_id.nil?
+      response = @api_resource.get("subnets/#{subnet_id}/")
+      json_body = JSON.parse(response.body)
+
+      data = {
+        id: json_body['data']['id'],
+        subnet: json_body['data']['subnet'],
+        mask: json_body['data']['mask'],
+        description: json_body['data']['description'],
+      }
+
+      return data if json_body['data']
+    end
+
+    def get_ipam_subnet_by_cidr(cidr)
+      subnet = @api_resource.get("subnets/cidr/#{cidr}")
+      json_body = JSON.parse(subnet.body)
+      return nil if json_body['data'].nil?
+
+      data = {
+        id: json_body['data'][0]['id'],
+        subnet: json_body['data'][0]['subnet'],
+        mask: json_body['data'][0]['mask'],
+        description: json_body['data'][0]['description'],
+      }
+
+      return data if json_body['data']
+    end
+
+    def get_ipam_group(group_name)
+      return nil if group_name.nil?
+      group = @api_resource.get("sections/#{group_name}/")
+      json_body = JSON.parse(group.body)
+      raise ERRORS[:no_group] if json_body['data'].nil?
+
+      data = {
+        id: json_body['data']['id'],
+        name: json_body['data']['name'],
+        description: json_body['data']['description'],
+      }
+
+      return data if json_body['data']
+    end
+
+    def ipam_groups
+      groups = @api_resource.get('sections/')
+      json_body = JSON.parse(groups.body)
+      return [] if json_body['data'].nil?
+
+      data = []
+      json_body['data'].each do |group|
+        data.push(
+          {
+            id: group['id'],
+            name: group['name'],
+            description: group['description'],
+          }
+        )
+      end
+
+      return data if json_body['data']
```

Comment:
You already have a guard earlier so the conditional for the return is redundant. You can also map it to transform it,removing the need for the additional variable.
```suggestion
      json_body['data'].map do |group|
        {
          id: group['id'],
          name: group['name'],
          description: group['description'],
        }
      end
```

Note that this is a repeated pattern in various of these methods.